lib/util.js

```diff
@@ -836,25 +853,69 @@ function formatMap(ctx, value, recurseTimes, keys) {
   return output;
 }
 
-function formatCollectionIterator(preview, ctx, value, recurseTimes,
-                                  visibleKeys, keys) {
-  const nextRecurseTimes = recurseTimes === null ? null : recurseTimes - 1;
-  const vals = preview(value, 100);
+function formatWeakSet(ctx, value, recurseTimes, keys) {
+  const maxArrayLength = Math.max(ctx.maxArrayLength, 0);
+  const entries = previewWeakSet(value, maxArrayLength + 1);
+  const maxLength = Math.min(maxArrayLength, entries.length);
+  let output = new Array(maxLength);
+  for (var i = 0; i < maxLength; ++i)
+    output[i] = formatValue(ctx, entries[i], recurseTimes);
+  // Sort all entries to have a halfway reliable output (if more entries than
+  // retrieved ones exist, we can not reliably return the same output).
+  output = output.sort();
+  if (entries.length > maxArrayLength)
+    output.push('... more items');
```

Comment:
Nit: the corresponding code in `formatArray()` is:

https://github.com/nodejs/node/blob/982e3bdb1f06bf9d9926c808d30864b32a8223f9/lib/util.js#L771-L772

We could do that instead.

Reply:
I do not see how that is possible. We have to explicitly enter a maximum number of entries we want. This is a upper bound and to be able to show that there are more entries, I actually get one more than requested with the regular maximum. We would have to get all entries each time (by passing through e.g. the maximum number of entries a Array may have) but that would be a significant overhead.

Reply:
I could get two more items to show "1 more item" and "more items".

Reply:
Ah I misread it then. LGTM!